src/prefect/environments/execution/cloud/environment.py

```diff
@@ -0,0 +1,200 @@
+
+    *Note*: This environment is not currently customizable. This may be subject to change.
+
+    There are no set up requirements, and `execute` creates a single job that has the role
+    of spinning up a dask executor and running the flow. The job created in the execute
+    function does have the requirement in that it needs to have an `identifier_label`
+    set with a UUID so resources can be cleaned up independently of other deployments.
+    """
+
+    def __init__(self) -> None:
+        self.identifier_label = str(uuid.uuid4())
+
+    def execute(self, storage: "Docker" = Docker()) -> None:  # type: ignore
+        """
+        Create a single Kubernetes job that spins up a dask scheduler, dynamically
+        creates worker pods, and runs the flow.
+
+        Args:
+            - storage (Docker): the Docker storage object that contains information relating
+                to the image which houses the flow
+        """
+        if not isinstance(storage, Docker):
+            raise TypeError("CloudEnvironment requires a Docker storage option")
+
+        if (
+            not storage.image_name
+            or not storage.image_tag
+            or not storage.registry_url
+            or not storage.flow_file_path
+        ):
+            raise ValueError("Docker storage is missing required fields")
+
+        self.create_flow_run_job(
+            registry_url=storage.registry_url,
+            image_name=storage.image_name,
+            image_tag=storage.image_tag,
+            flow_file_path=storage.flow_file_path,
+        )
+
+    def create_flow_run_job(
+        self, registry_url: str, image_name: str, image_tag: str, flow_file_path: str
+    ) -> None:
+        """
+        Creates a Kubernetes job to run the flow using the information stored on the
+        Docker storage object.
+
+        Args:
+            - registry_url (str): URL of a registry the image was stored in
+            - image_name (str): name of the image
+            - image_tag (str): tag of the image
+            - flow_file_path (str): location of the flow file in the image
+        """
+        from kubernetes import client, config
+
+        # Verify environment is running in cluster
+        try:
+            config.load_incluster_config()
+        except config.config_exception.ConfigException:
+            raise EnvironmentError("Environment not currently inside a cluster")
+
+        batch_client = client.BatchV1Api()
+
+        with open(path.join(path.dirname(__file__), "job.yaml")) as job_file:
+            job = yaml.safe_load(job_file)
+            job = self._populate_job_yaml(
+                yaml_obj=job,
+                registry_url=registry_url,
+                image_name=image_name,
+                image_tag=image_tag,
+                flow_file_path=flow_file_path,
+            )
+
+            # Create Job
+            batch_client.create_namespaced_job(namespace="default", body=job)
+
+    def run_flow(self) -> None:
```

Comment:
Should this be changed to `run()` to match command that you submit in the `yml` file below?